apps/settings/js/panels/findmydevice/findmydevice.js

```diff
@@ -39,48 +58,58 @@ define('findmydevice', ['modules/settings_utils', 'shared/settings_listener'
           onlogin: self._onChangeLoginState.bind(self, true),
           onlogout: self._onChangeLoginState.bind(self, false),
           onready: function fmd_fxa_onready() {
-            self._loginButton.removeAttribute('disabled');
+            self.elements.loginButton.removeAttribute('disabled');
             console.log('Find My Device: onready fired');
-          },
+          }.bind(self),
           onerror: function fmd_fxa_onerror(err) {
             console.error('Find My Device: onerror fired: ' + err);
             self._interactiveLogin = false;
-            self._loginButton.removeAttribute('disabled');
+            self.elements.loginButton.removeAttribute('disabled');
             var errorName = JSON.parse(err).name;
             if (errorName !== 'OFFLINE') {
               if (errorName === 'UNVERIFIED_ACCOUNT') {
-                var unverifiedError = document.getElementById(
-                  'findmydevice-fxa-unverified-error');
-                unverifiedError.hidden = false;
-                var login = document.getElementById('findmydevice-login');
-                login.hidden = true;
+                self.elements.unverifiedError.hidden = false;
+                self.elements.login.hidden = true;
               }
-
               self._loggedInHelper.set(false);
             }
           }
         });
       });
-
+    },
+    onBeforeShow: function fmd_onBeforeShow() {
       SettingsListener.observe('findmydevice.tracking', false,
         this._setTracked.bind(this));
```

Comment:
please use `_boundFunc... = func().bind(this)` convention to make sure we do handle the same object